app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt

```diff
@@ -9,24 +9,34 @@ import mozilla.components.browser.state.selector.selectedNormalTab
 import mozilla.components.browser.state.state.BrowserState
 import mozilla.components.browser.state.state.TabSessionState
 import mozilla.components.feature.tabs.ext.hasMediaPlayed
+import org.mozilla.fenix.FeatureFlags
+import org.mozilla.fenix.home.recenttabs.RecentTab
+import org.mozilla.fenix.tabstray.browser.TabGroup
+import org.mozilla.fenix.tabstray.browser.maxActiveTime
+import org.mozilla.fenix.tabstray.ext.isNormalTabActiveWithSearchTerm
+import kotlin.math.max
 
 /**
  * Get the last opened normal tab and the last tab with in progress media, if available.
  *
- * @return A list of the last opened tab and the last tab with in progress media
+ * @return An array list of the last opened tab and the last tab with in progress media
  * if distinct and available or an empty list.
  */
-fun BrowserState.asRecentTabs(): List<TabSessionState> {
-    return mutableListOf<TabSessionState>().apply {
+fun BrowserState.asRecentTabs(): ArrayList<RecentTab> {
```

Comment:
Why change it to an `ArrayList`?

Reply:
This makes it easy to convert to an array to use in `RecentTabsListFeature` when we listen for changes in `ifAnyChanged`

Reply:
I'll change this back to List now that you've shown be a better way to listen for changes (see comment below). Thanks,